dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -534,6 +534,10 @@ private boolean processComplementData() {
             scheduleDate = complementListDate.get(0);
         } else if (processInstance.getState().typeIsFinished()) {
             endProcess();
+            // rerun process instance of complement didn't need create the next process complement
+            if (processInstance.getCommandType() == CommandType.REPEAT_RUNNING) {
+                return true;
+            }
```

Comment:
Discusstion in https://github.com/apache/dolphinscheduler/issues/12586#issuecomment-1296455628

Reply:
It's a feature, not a bugfix, should not add into 2.0.x version.